pkg/downstream/suggest.go

```diff
@@ -64,28 +67,26 @@ func Suggest(ctx context.Context, git utils.GitHelper, client *github.Client, re
 			if len(msgLines) == 0 {
 				return fmt.Errorf("found commit with empty body: %s", c.GetSHA())
 			}
-			out, err := git.DoOutput("log", "--fixed-strings", "--grep", msgLines[0])
+			out, err := git.DoOutput("log", "--fixed-strings", "--pretty=format:%h", "--grep", msgLines[0])
 			if err != nil {
 				return err
 			}
-			if len(out) > 0 {
+			found := strings.Split(out, "\n")
+			hasCommit, err := hasCommit(git, req, found, c.GetCommit())
+			if err != nil {
+				return err
+			}
+			if hasCommit {
 				numFoundCommits++
 			}
 			numCommits++
 		}
 
-		// it may happen that a PR has been partially downstreamed in
-		// the fork and some of its commits have been discarded. In such
-		// a case, we'll not suggest it as a dowstream candidate but we'll
-		// log it for sake of transparency
-		if numFoundCommits > 0 && numFoundCommits < numCommits {
-			logrus.Warningf("pull request %d has been partially ported (%d/%d commits): %s", v.GetNumber(), numFoundCommits, numCommits, v.GetHTMLURL())
-		}
-
-		// if none of the PR's commit are present in the downstream fork
+		// if less than the 50% of the PR's commit are present in the downstream fork
 		// history (checked from the provided head), then we can conclude
```

Comment:
Let's not remove the warning log though, otherwise it would be impossible to understand from the log why we discarded the PR/commit